projects/i18n/src/lib/pipe/translation-pipe.ts

```diff
@@ -46,9 +47,12 @@ export class TranslationPipe implements PipeTransform {
     }
 
     private updateValue(key: string, ...args: any[]): void {
-        this.translate.translateAsync(key, args).subscribe(result => {
-            this.value = result;
-            this.changeDetector.markForCheck();
-        });
+        this.translate
```

Comment:
Calling `toPromise()` seems like a more straight forward way of getting values from an observable without holding on to a subscription.

```ts
const result = await this.translate.translateAsync(key, args).toPromise();
this.value = result;
this.changeDetector.markForCheck();
```